lib/oauthutil/oauthutil.go

```diff
@@ -470,9 +472,13 @@ func doConfig(id, name string, m configmap.Mapper, oauthConfig *oauth2.Config, o
 		authURL = "http://" + bindAddress + "/auth?state=" + state
 	}
 
-	// Open the URL for the user to visit
-	_ = open.Start(authURL)
-	fmt.Printf("If your browser doesn't open automatically go to the following link: %s\n", authURL)
+	if !authorizeNoAutoBrowser && oauthConfig.RedirectURL != TitleBarRedirectURL {
```

Comment:
Why does this have `&& oauthConfig.RedirectURL != TitleBarRedirectURL` ?

`TitleBarRedirectURL` still needs the browser window open doesn't it?

Reply:
Well this is actually a complicated logic :)

There are two scenarios that will open the browser,
- `rclone authorize` with no option given
- `rclone config`, and use auto config

In the second scenarios, if the `oauthConfig.RedirectURL` was `TitleBarRedirectURL`, and user chooses to use auto config, it will be changed to the internal web server instead of `TitleBarRedirectURL`. So if in this line `oauthConfig.RedirectURL == TitleBarRedirectURL`, it means the user has choosen not to use auto config, and not open the browser automatically.